src/wovnio/wovnphp/custom_domain/CustomDomainLangs.php

```diff
@@ -1,22 +1,39 @@
 <?php
 namespace Wovnio\Wovnphp;
 
+use PHP_CodeSniffer\Tests\Standards\AllSniffs;
+
 require_once 'CustomDomainLang.php';
 
 class CustomDomainLangs
 {
     private $customDomainLangs;
+
+
     public function __construct($customDomainLangsSettingsArray)
     {
         $this->customDomainLangs = array();
-        foreach ($customDomainLangsSettingsArray as $langUrl => $lang) {
+        foreach ($customDomainLangsSettingsArray as $langUrl => $config) {
             $parsedUrl = parse_url($this->addProtocolIfNeeded($langUrl));
-
+            $source = array_key_exists('source', $config) ? $config['source'] : null;
             // Disable notice error by adding @, when path is not defined
-            array_push($this->customDomainLangs, new CustomDomainLang($parsedUrl['host'], @$parsedUrl['path'], $lang));
+            array_push($this->customDomainLangs, new CustomDomainLang($parsedUrl['host'], @$parsedUrl['path'], $config['lang'], $source));
         }
     }
 
+    public function getSourceCustomDomainByLang($langCode)
```

Comment:
How i'm expecting this to work is:
 -Every language can have a source associated with it meaning where the raw HTML comes from
- if a given language does not have a source specified for it, default to the default language's source.

Would like to avoid "bypass" in our naming if possible. Physical URL would be preferred.

Also avoid using this->settings['custom_domain_langs']. Everything should be getting delegated through the CustomDomainLangs class.